### PR TITLE
Add SNNMLP to pytorch-hub

### PR DESCRIPTION
**Pull request by Gaffey**
SNNMLP incorporates the mechanism of LIF neurons into the MLP models, to achieve better accuracy without extra FLOPs. We propose a full-precision LIF operation to communicate between patches, including horizontal LIF and vertical LIF in different directions. We also propose to use group LIF to extract better local features. With LIF modules, our SNNMLP model achieves 81.9%, 83.3% and 83.6% top-1 accuracy on ImageNet dataset with only 4.4G, 8.5G and 15.2G FLOPs, respectively.

The corresponding accuracy on ImageNet dataset with pretrained model is listed below.

| Model structure | #Parameters | FLOPs       | Top-1 acc   |
| --------------- | ----------- | ----------- | ----------- |
|  SNNMLP Tiny    | 28M         | 4.4G        | 81.88       |
|  SNNMLP Small   | 50M         | 8.5G        | 83.30       |
|  SNNMLP Base    | 88M         | 15.2G       | 85.59       |

The full paper could be found at this [link](https://arxiv.org/abs/2203.14679).

### CODE REVIEW

File: pytorch_vision_snnmlp.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+# Tensor of shape 1000, with confidence scores over Imagenet's 1000 classes
+print(output[0])
+# The output has unnormalized scores. To get probabilities, you can run a softmax on it.
+print(torch.nn.functional.softmax(output[0], dim=0))
+
+```
+
+### Model Description
+
+SNNMLP incorporates the machanism of LIF neurons into the MLP models, to achieve better accuracy without extra FLOPs. We propose a full-precision LIF operation to communicate between patches, including horizontal LIF and vertical LIF in different directions. We also propose to use group LIF to extract better local features. With LIF modules, our SNNMLP model achieves 81.9%, 83.3% and 83.6% top-1 accuracy on ImageNet dataset with only 4.4G, 8.5G and 15.2G FLOPs, respectively.
```
Comment (vmoens) [inline pytorch_vision_snnmlp.md:74]: machanism -> mechanism

File: pytorch_vision_snnmlp.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+| Model structure | #Parameters | FLOPs       | Top-1 acc   |
+| --------------- | ----------- | ----------- | ----------- |
+|  SNNMLP Tiny    | 28M         | 4.4G        | 81.88       |
+|  SNNMLP Small   | 50M         | 8.5G        | 83.30       |
+|  SNNMLP Base    | 88M         | 15.2G       | 85.59       |
+
+
+### References
+
+You can read the full paper at this [link](https://arxiv.org/abs/2203.14679).
```
Comment (vmoens) [inline pytorch_vision_snnmlp.md:87]: what about ```You can read the full paper [here](https://arxiv.org/abs/2203.14679)```

**Review by vmoens — APPROVED** [review]
LGTM, just a couple of minor changes and we're good to go.
Thanks a million for contributing!

**Review by vmoens — APPROVED** [review]
Great! Let's merge!
Thanks for your contribution!
